apps/mobile/src/features/Discover/common/useDAppsConnected.tsx

```diff
@@ -1,48 +1,47 @@
-import {useQuery, useQueryClient, UseQueryOptions} from '@tanstack/react-query'
 import {DappConnection, useDappConnector} from '@yoroi/dapp-connector'
 import {Chain} from '@yoroi/types'
 import * as React from 'react'
 
 import {useSelectedWallet} from '~/features/WalletManager/hooks/useSelectedWallet'
+import {usePromise} from '~/hooks/usePromise'
 
-export const useDAppsConnected = (
-  options?: UseQueryOptions<
-    DappConnection[],
-    Error,
-    string[],
-    [string, string, string]
-  >,
-) => {
+export const useDAppsConnected = (options?: {refetchOnMount?: boolean}) => {
   const {wallet} = useSelectedWallet()
   const {manager} = useDappConnector()
 
-  return useQuery({
-    suspense: true,
+  const result = usePromise({
```

Comment:
@jorbuedo usePromise works like useFocusEffect, if a the dependency can change, it needs to be wrapped in a React.useCallback(, [thingsThatCanChange]) rabbitai got this one right.  :D